src/components/shared/transactionsTable/index.js

```diff
@@ -1,208 +1,229 @@
-              {t('LSK')}
-            </div>
-          </IconlessTooltip>
-        );
-      case 'timestamp':
-        return <DateTimeFromTimestamp time={transaction[column.key] * 1000} token="BTC" />;
-      case 'confirmations':
-        return (
-          <IconlessTooltip
-            tooltipContent={<p>{`${transaction.confirmations}/101 ${t('Confirmations')}`}</p>}
-            title={transaction.confirmations > 0 ? t('Confirmed') : t('Pending')}
-            className="showOnLeft"
-            tooltipClassName={styles.tooltip}
-          >
-            {transaction.confirmations > 0 ? <Icon name="approved" /> : <Icon name="pending" />}
-          </IconlessTooltip>
-        );
-      default:
-        return transaction[column.key];
-    }
-  }
-
-  // TODO add test when sorting is enabled
-  // istanbul ignore next
-  changeSorting(column) {
-    if (column.isSortingColumn) {
-    // Only changes the caret.
-    // TODO: Implementation of sorting functionality to be done in a separate ticket.
-      this.setState({
-        sortingColumn: column.key,
-        ascendingSorting:
-        column.key === this.state.sortingColumn ? !this.state.ascendingSorting : true,
-      });
-    }
+    transactions.loadData(Object.keys(filters).reduce((acc, key) => ({
+      ...acc,
+      ...(filters[key] && { [key]: filters[key] }),
+    }), {
+      offset: transactions.data.length,
+      sort,
+    }));
   }
 
-  // TODO add test when sorting is enabled
-  // istanbul ignore next
-  renderTransactions() {
-    const { sortingColumn } = this.state;
+  getTransformedAddress(address) {
+    const { isMediumViewPort, bookmarks } = this.props;
+    const bookmarkedAddress = bookmarks.LSK.find(element => element.address === address);
 
-    return this.props.transactions.data.sort((a, b) =>
-      (this.state.ascendingSorting
-        ? b[sortingColumn] - a[sortingColumn]
-        : a[sortingColumn] - b[sortingColumn]));
+    if (bookmarkedAddress) return bookmarkedAddress.title;
+    if (isMediumViewPort) return address.replace(regex.lskAddressTrunk, '$1...$3');
+    return address;
   }
 
-  loadMore() {
-    const { transactions } = this.props;
-    transactions.loadData({ offset: transactions.data.length });
+  renderAddressField(transaction, key) {
+    const { t } = this.props;
+
+    if (transaction.type !== 0 && key === 'recipientId') {
+      return (
+        <div className={`${styles.address}`}>
+          <Icon
+            className={styles.txIcon}
+            name={transactionTypeIcons[transaction.type] || transactionTypeIcons.default}
+          />
+          <span className={styles.addressValue}>{transactionNames(t)[transaction.type]}</span>
+        </div>
+      );
+    }
+    return (
+      <div className={`${styles.address}`}>
+        <AccountVisual address={transaction[key]} size={32} />
+        <span className={styles.addressValue}>
+          {this.getTransformedAddress(transaction[key])}
+        </span>
+      </div>
+    );
```

Comment:
This is a good candidate for a shared component, as we'll have the same in other pages, e.g.:
https://projects.invisionapp.com/d/main/#/console/18539694/386733858/inspect

The name could be something like `<AccountVisualWithAddress />`

Reply:
Good idea. I've created the shared component `AccountVisualWithAddress` and moved this logic there. I've made the replacement of the address with the bookmarked name optional there, since I noticed the Delegates tab already has the `Name` column and, if bookmarked, that name would appear twice.